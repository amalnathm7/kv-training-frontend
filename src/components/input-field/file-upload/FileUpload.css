.file-upload {
    width: 30%;
    padding: 20px;
}

.file-upload-field {
    margin-top: 15px;
    align-items: center;
    justify-content: center;
    width: 100%;
    border-radius: 5px;
    padding: 10px;
    padding-top: 16.5px;
    padding-bottom: 16.5px;
    caret-color: #03AEEE;
    border: 2px solid lightgray;
    outline: 0;
    background: #FAFAFA 0% 0% no-repeat padding-box;
    border: 1px solid #D9D9D9;
}

.file-upload-label {
    margin-top: 20px;
    font-size: 14px;
    color: #555555;
    font-weight: bold;
}

.file-upload-error {
    margin-top: 10px;
    text-indent: 10px;
    font-size: 14px;
    color: red;
}