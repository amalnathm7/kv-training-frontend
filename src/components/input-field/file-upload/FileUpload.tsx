import React from 'react';
import './FileUpload.css';

export type FileUploadPropsType = {
  onChange: (event) => void;
  label: string;
  showError: boolean;
};

const FileUpload: React.FC<FileUploadPropsType> = (props) => {
  return (
    <div className='file-upload'>
      <label className='file-upload-label'>{props.label}</label>
      <input
        className='file-upload-field'
        type='file'
        name='filename'
        onChange={props.onChange}
      ></input>
      {props.showError && (
        <label className='file-upload-error'>Enter {props.label.toLowerCase()}</label>
      )}
    </div>
  );
};

export default FileUpload;
