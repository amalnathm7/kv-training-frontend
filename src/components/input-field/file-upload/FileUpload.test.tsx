import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import FileUpload, { FileUploadPropsType } from './FileUpload';

describe('File upload Props Test', () => {
  test('If file upload field rendered correctly', () => {
    const props: FileUploadPropsType = {
      onChange: () => {},
      label: 'file-upload-label',
      showError: false
    };

    const element = render(<FileUpload {...props} />);

    expect(element).toMatchSnapshot();
  });

  test('If file upload field rendered correctly with error', () => {
    const props: FileUploadPropsType = {
      onChange: () => {},
      label: 'file-upload-label',
      showError: true
    };

    const element = render(<FileUpload {...props} />);

    expect(element).toMatchSnapshot();
  });
});
