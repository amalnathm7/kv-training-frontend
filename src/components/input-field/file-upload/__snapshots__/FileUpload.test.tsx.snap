// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File upload Props Test If file upload field rendered correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="file-upload"
      >
        <label
          class="file-upload-label"
        >
          file-upload-label
        </label>
        <div
          class="file-upload-field"
        >
          <input
            id="file-upload-input"
            type="file"
          />
          <label
            class="file-upload-value"
            for="file-upload-input"
          >
            Select resume	
          </label>
          <label
            class="file-upload-value-rules"
            for="file-upload-input"
          >
            File type: PDF, Max size: 5 MB
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="file-upload"
    >
      <label
        class="file-upload-label"
      >
        file-upload-label
      </label>
      <div
        class="file-upload-field"
      >
        <input
          id="file-upload-input"
          type="file"
        />
        <label
          class="file-upload-value"
          for="file-upload-input"
        >
          Select resume	
        </label>
        <label
          class="file-upload-value-rules"
          for="file-upload-input"
        >
          File type: PDF, Max size: 5 MB
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`File upload Props Test If file upload field rendered correctly with error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="file-upload"
      >
        <label
          class="file-upload-label"
        >
          file-upload-label
        </label>
        <div
          class="file-upload-field"
        >
          <input
            id="file-upload-input"
            type="file"
          />
          <label
            class="file-upload-value"
            for="file-upload-input"
          >
            Select resume	
          </label>
          <label
            class="file-upload-value-rules"
            for="file-upload-input"
          >
            File type: PDF, Max size: 5 MB
          </label>
        </div>
        <label
          class="file-upload-error"
        >
          Enter valid 
          file-upload-label
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="file-upload"
    >
      <label
        class="file-upload-label"
      >
        file-upload-label
      </label>
      <div
        class="file-upload-field"
      >
        <input
          id="file-upload-input"
          type="file"
        />
        <label
          class="file-upload-value"
          for="file-upload-input"
        >
          Select resume	
        </label>
        <label
          class="file-upload-value-rules"
          for="file-upload-input"
        >
          File type: PDF, Max size: 5 MB
        </label>
      </div>
      <label
        class="file-upload-error"
      >
        Enter valid 
        file-upload-label
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`File upload Props Test If file upload field rendered correctly with file value 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="file-upload"
      >
        <label
          class="file-upload-label"
        >
          file-upload-label
        </label>
        <div
          class="file-upload-field"
        >
          <input
            id="file-upload-input"
            type="file"
          />
          <label
            class="file-upload-value"
            for="file-upload-input"
          >
            foo.txt
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="file-upload"
    >
      <label
        class="file-upload-label"
      >
        file-upload-label
      </label>
      <div
        class="file-upload-field"
      >
        <input
          id="file-upload-input"
          type="file"
        />
        <label
          class="file-upload-value"
          for="file-upload-input"
        >
          foo.txt
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`File upload Props Test If file upload field rendered correctly with value 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="file-upload"
      >
        <label
          class="file-upload-label"
        >
          file-upload-label
        </label>
        <div
          class="file-upload-field"
        >
          <input
            id="file-upload-input"
            type="file"
          />
          <label
            class="file-upload-value"
            for="file-upload-input"
          >
            file-upload-value
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="file-upload"
    >
      <label
        class="file-upload-label"
      >
        file-upload-label
      </label>
      <div
        class="file-upload-field"
      >
        <input
          id="file-upload-input"
          type="file"
        />
        <label
          class="file-upload-value"
          for="file-upload-input"
        >
          file-upload-value
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
